packages/eslint-plugin/src/rules/no-shadow.ts

```diff
@@ -513,15 +520,30 @@ export default createRule<Options, MessageIds>({
       const inner = getNameRange(variable);
       const outer = getNameRange(scopeVar);
 
-      return !!(
-        inner &&
-        outer &&
-        inner[1] < outer[0] &&
-        // Excepts FunctionDeclaration if is {"hoist":"function"}.
-        (options.hoist !== 'functions' ||
-          !outerDef ||
-          outerDef.node.type !== AST_NODE_TYPES.FunctionDeclaration)
-      );
+      if (!inner || !outer || inner[1] >= outer[0]) {
+        return false;
+      }
+
+      if (!outerDef) {
+        return true;
+      }
```

Comment:
I went over this again and made one small change. I think I've introduced an unnecessary change to the original implementation (returning `false` instead of `true` if `outerDef` is `false`).

I don't think this code is reachable (all tests seem to pass regardless of this change), and I couldn't reproduce this in a test. Still, just to be sure, I've changed this to match the original implementation.

Reply:
Now that it's on its own `if` statement, it fails codecov (even though this wasn't covered previously but was in a single long binary operation). I'm a bit unsure what the best way to tackle this is. I'd normally use `nullThrows` or a type assertion, but I'm not entirely sure this really is an unreachable line.

Reply:
Yeah I'd just ignore Codecov here. Its reporter isn't perfect.